### PR TITLE
[VP9D]  show_existing_frame with DRC

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
1) With MFX_MEMTYPE_EXTERNAL_FRAME and MFX_MEMTYPE_DXVA2_DECODER_TARGET
copying MFX_MEMTYPE_EXTERNAL_FRAME frames fails without locking frame
2) We should re-allocate internal D3D decoding surface in same way as external output surface in case of REALLOC_SURFACE 

MDP-50343

Test:
./mfx_player -i Stress_VP9_FC2p1ss444_1920x1080_250_extra_stress_2.2.0.vp9 -o:i444 -hw -d3d -async 5 -crc -async_depth 5 -vp9_drc

./mfx_player -i Stress_VP9_FC2p1ss444_1920x1080_250_extra_stress_2.2.0.vp9 -o:i444 -hw -sys -async 5 -crc -async_depth 5 -vp9_drc -gpu_copy 1 -adaptive_playback

### CODE REVIEW

File: _studio/shared/src/libmfx_core_vaapi.cpp [hunk 1]
```diff
@@ -897,8 +897,10 @@ VAAPIVideoCORE::DoFastCopyWrapper(
 
     if (srcMemType & MFX_MEMTYPE_EXTERNAL_FRAME)
     {
+#ifndef __linux__
```
Comment (dmitryermilov) [inline _studio/shared/src/libmfx_core_vaapi.cpp]: please remove this and below debug code

File: _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp [hunk 2]
```diff
@@ -871,9 +892,13 @@ mfxStatus VideoDECODEVP9_HW::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
...
     {
-        repeateFrame = m_frameInfo.ref_frame_map[m_frameInfo.frame_to_show];
-        m_FrameAllocator->IncreaseReference(repeateFrame);
+        if (UMC::UMC_OK != m_FrameAllocator->IncreaseReference(repeatedFrame))
+        {
+            return MFX_ERR_UNKNOWN;
+        }
         ++m_statusReportFeedbackNumber;
+        //surface_work->Info.CropW = (mfxU16)m_frameInfo.width;
+        //surface_work->Info.CropH = (mfxU16)m_frameInfo.height;
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp]: Please clean up

File: _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp [hunk 3]
```diff
@@ -574,9 +574,18 @@ void VideoDECODEVP9_HW::UpdateVideoParam(mfxVideoParam *par, VP9DecoderFrame con
...
-    if (   par->mfx.FrameInfo.FourCC == MFX_FOURCC_P010
-        )
+    if (par->mfx.FrameInfo.FourCC == MFX_FOURCC_P010)
         par->mfx.FrameInfo.Shift = 1;
+
+    if (frameInfo.show_existing_frame)
+    {
+        UMC::FrameMemID repeatedFrame = m_frameInfo.ref_frame_map[m_frameInfo.frame_to_show];
+        mfxFrameSurface1 *repeatedSurface = m_FrameAllocator->GetSurfaceByIndex(repeatedFrame);
+        par->mfx.FrameInfo.Width = repeatedSurface->Info.Width;
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp]: please add a comment why we re-write current params here

File: _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp [hunk 4]
```diff
@@ -830,8 +841,20 @@ mfxStatus VideoDECODEVP9_HW::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
...
     UMC::ColorFormat const cf = GetUMCColorFormat_VP9(&m_frameInfo);
-    if (UMC::UMC_OK != videoInfo.Init(m_vPar.mfx.FrameInfo.Width, m_vPar.mfx.FrameInfo.Height, cf, m_frameInfo.bit_depth))
-        return MFX_ERR_MEMORY_ALLOC;
+
+    if (m_frameInfo.show_existing_frame)
+    {
+        repeatedFrame = m_frameInfo.ref_frame_map[m_frameInfo.frame_to_show];
+        repeatedSurface = m_FrameAllocator->GetSurfaceByIndex(repeatedFrame);
+
+        if (UMC::UMC_OK != videoInfo.Init(repeatedSurface->Info.Width, repeatedSurface->Info.Height, cf, m_frameInfo.bit_depth))
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp]: it's possible to avoid code duplication at 850 and 855

**Comment by IlyaSergeyev** [comment]
All resolution corrections was removed from initial code. The root of the bug is copying frames without locking on Linux.

**Comment by mgonchar** [comment]
Looks like something wrong with your commits history

**Comment by IlyaSergeyev** [comment]
Closed in favor of #1140

**Pull request closed without merging**